Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3800 @@
+                raise DemistoException("Auth Token Expired, Refresh Token is not found. please create a new Auth Token.")
+
+    return None
+
+
+def create_client_header(
+    use_token: bool,
+    username: str,
+    password: str,
+    client_id: str,
+    client_secret: str,
+    access_token: str,
+):
+    """
+    Creates Auth and Header arguments for the Client object based on the authentication method selected in the integration instance configuration.
+
+    :type use_token: ``bool``
+    :param use_token: boolean argument indicating which type of authentication will be used.
+
+    :type username: ``str``
+    :param username: Username parameter from instance configuration.
+
+    :type password: ``str``
+    :param password: Password parameter from instance configuration.
+
+    :type client_id: ``str``
+    :param client_id: Client ID parameter from instance configuration.
+
+    :type client_secret: ``str``
+    :param client_secret: Client Secret parameter from instance configuration.
+
+    :type access_token: ``str``
+    :param access_token: Access Token parameter from instance configuration.
+
+    :rtype: Tuple(Optional[None,str],dict[str,str])
+    returns (auth, header) tuple
+    """
+    # Basic authentication
+    if not use_token and username and password:
+        credentials = f"{username}:{password}"
+        auth_header = f"Basic {b64_encode(credentials)}"
+        return (username, password), {"Authorization": auth_header}
+    # Auth Token authentication
+    elif use_token and all([client_id, client_secret, access_token]):
+        return None, {"Authorization": f"Bearer {access_token}"}
+    else:
+        raise DemistoException(message="Not all fields for the selected authenticating are set,therefor it cannot be executed.")
+
+
+# HEADING: """ COMMAND FUNCTIONS """
+
+
+def create_auth_client(client: Client, args: Dict[str, Any]) -> CommandResults:
+
+    if not client._auth:
+        raise DemistoException(
+            message="Auth process failed. To run vd-auth-start command, please enter Username and password parameters in instance configuration."
+        )
+    username, password = client._auth
+    token_name = args.get("name", "XSOAR-OAuth-client")
+    token_description = args.get("description", "XSOAR OAuth client for Versa Director Integration")
+    client_id = CLIENT_ID
+    client_secret = CLIENT_SECRET
+
+    if client._base_url.endswith(PORT_CREDENTIALS):
+        client._base_url = client._base_url.replace(PORT_CREDENTIALS, PORT_AUTH)
+    try:
+        token_response = client.access_token_request(username, password, client_id, client_secret)
+        access_token = token_response.get("access_token", "")
+
+        response = client.auth_credentials_request(access_token, token_name, token_description)
+
+    except DemistoException as e:
+        if e.res.status_code == 500:
+            raise DemistoException(message="Auth process failed. Existing Token found with the matching name.")
```

Comment:
Lets Maybe take all the errors and put it in const? It will make the lines more readable and maintainable.